org.eclipse.xtext.util/src/org/eclipse/xtext/util/formallang/NfaUtil.java

```diff
@@ -617,12 +617,12 @@ public void accept(List<S> t) {
 			}
 		};
 		
-		findCycles(nfa, nfa.getStart(), cycleAcceptor, Maps.<S, Integer>newHashMap(), Lists.<S>newLinkedList());
+		findCycles(nfa, nfa.getStart(), cycleAcceptor, Maps.<S, Integer>newLinkedHashMap(), Lists.<S>newLinkedList());
```

Comment:
Change is ok, but since the map is not exposed or traversed, it's probably not contributing to the fix.